lib/src/working_copy.rs

```diff
@@ -126,12 +126,12 @@ pub struct TreeState {
 }
 
 fn file_state_from_proto(proto: &crate::protos::working_copy::FileState) -> FileState {
```

Comment:
Hmm, here you didn't change it to pass `proto` by value. I'm not sure which is better, but it seems that we  should be consistent.

Reply:
General rule of thumb is that if you were going to clone everything, then you should take ownership instead so that the caller has the option to pass ownership to you and save a clone if they didn't need to hold onto the value. In this specific case, taking ownership is necessary is further useful because it lets us call the concise `unwrap_or_default()` rather than the `as_ref().cloned().unwrap_or_default()` we'd need otherwise.

I'll see about standardizing on that.

Reply:
Left a couple in working_copy.rs whose caller really did want to reuse the values, and which didn't take ownership of any fields anyway.

Reply:
Nice, that was a good cleanup.